packages/opentelemetry-sdk-trace-base/src/export/BatchSpanProcessorBase.ts

```diff
@@ -117,8 +118,23 @@ export abstract class BatchSpanProcessorBase<T extends BufferConfig>
   private _addToBuffer(span: ReadableSpan) {
     if (this._finishedSpans.length >= this._maxQueueSize) {
       // limit reached, drop span
+
+      if (this._droppedSpansCount === 0) {
+        diag.debug('maxQueueSize reached, dropping spans');
+      }
+      this._droppedSpansCount++;
```

Comment:
This will be _very_ chatty in a system overloaded enough to be dropping spans. Do you think it's needed when the warn log on line 132 is already doing this job?

Reply:
For a given stretch of dropped spans, this message will only be reported once, just like the summary warning, so should have the same cardinality as that one. This debug diagnostic can be used to identify at what time those drops started. It's recorded as debug, which should be able to be filtered out if debug diagnostics aren't interesting or desired.

I found it useful as there are two knobs that can be adjusted to prevent dropping spans: queue size and the export interval. The warning message makes it easier to adjust the queue size, but the debug message can make it apparent how often one should be exporting if they don't want to carry a larger memory load associated with a larger queue.

I'm not particularly attached to this message, so if we want it removed, that's fine with me.